snabble/Android-SDK
# Update user agent headers (Apps 727)

### What's new?

The old user agend header as been slighty adjustes and new headers have been added.
The new format were taken from [iOS](https://github.com/snabble/snabble-ios-sdk/blob/12767c7dfda7d952007d2fa93c58dbab6131250a/Sources/Core/API/Snabble.swift#L626)

### How to test?
Launch the app and apply this patch (see headers in the logs):
```diff
diff --git a/core/src/main/java/io/snabble/sdk/auth/useragent/UserAgentInterceptor.kt b/core/src/main/java/io/snabble/sdk/auth/useragent/UserAgentInterceptor.kt
index 1dae4a85c..0d1a681a1 100644
--- a/core/src/main/java/io/snabble/sdk/auth/useragent/UserAgentInterceptor.kt
+++ b/core/src/main/java/io/snabble/sdk/auth/useragent/UserAgentInterceptor.kt
@@ -1,6 +1,7 @@
 package io.snabble.sdk.auth.useragent
 
 import android.content.Context
+import io.snabble.sdk.extensions.xx
 import okhttp3.Interceptor
 import okhttp3.Request
 import okhttp3.Response
@@ -20,6 +21,8 @@ internal class UserAgentInterceptor(private val context: Context) : Interceptor
 
         val finalRequest = userAgentRequestBuilder.build()
 
+        finalRequest.headers.xx()
+
         return chain.proceed(finalRequest)
     }
 }
```

### Definition of Done

#### App Tests
- [ ] Environements beachtet _(Production/Staging)_
- [ ] Unterstützte Sprachen sind getestet
- [ ] Light-/Dark-Mode getestet

####Testing
- [ ] Tests geschrieben _(aka Unit-Tests, Integration-Tests)_
